### PR TITLE
Add support for geometry tasks in submission details view

### LINKED ISSUE #1367: [Submission details] Show geometry responses on map for selected submission
Shown when a submission is selected from the submissions list (#1131)

image

### PR DESCRIPTION
**Pull request by amysorto**
Fixes #1367

[Geometry tasks.webm](https://github.com/google/ground-platform/assets/8575252/6671b0ce-3d77-45fb-8443-d5cb787058f1)

### CODE REVIEW

**Comment by gino-m** [comment]
Hi @amysorto! It looks like the geometry annotations (geometry task responses, non-LOI markers and polygons on the map) don't show their corresponding number shown in the panel. Want to file a TODO and a bug to do in a follow-up?

See designs here:

image

File: web/src/app/pages/main-page-container/main-page/map/map.component.ts [hunk 1]
```diff
@@ -166,6 +175,76 @@ export class MapComponent implements AfterViewInit, OnDestroy {
         this.navigationService.getSubmissionId$(),
       ]).subscribe(() => this.cancelReposition())
     );
+
+    this.subscription.add(
+      this.submissionService.getSelectedSubmission$().subscribe(submission => {
```
Comment (gino-m) [inline web/src/app/pages/main-page-container/main-page/map/map.component.ts:180]: Would this cause a race condition with the `submissionId` and `locationOfInterestId` above?

File: web/src/app/pages/main-page-container/main-page/map/map.component.ts [hunk 2]
```diff
@@ -166,6 +175,76 @@ export class MapComponent implements AfterViewInit, OnDestroy {
...
+          if (this.showSubmissionGeometry === true) {
+            this.removeSubmissionResultsOnMap();
+            this.addSubmissionResultsOnMap();
+          }
+        }
+      })
+    );
+
+    this.subscription.add(
+      this.navigationService.getSubmissionId$().subscribe(id => {
```
Comment (gino-m) [inline web/src/app/pages/main-page-container/main-page/map/map.component.ts:192]: We already subscribe to changes to submissionId above, perhaps this should live there?

File: web/src/app/pages/main-page-container/main-page/map/map.component.ts [hunk 3]
```diff
@@ -166,6 +175,76 @@ export class MapComponent implements AfterViewInit, OnDestroy {
...
+        this.showSubmissionGeometry = id ? true : false;
```
Comment (gino-m) [inline web/src/app/pages/main-page-container/main-page/map/map.component.ts:193]: Is this equivalent to`!!id`?

File: web/src/app/pages/main-page-container/main-page/map/map.component.ts [hunk 4]
```diff
@@ -166,6 +175,76 @@ export class MapComponent implements AfterViewInit, OnDestroy {
...
+        this.removeSubmissionResultsOnMap();
+        if (this.showSubmissionGeometry === true) {
```
Comment (gino-m) [inline web/src/app/pages/main-page-container/main-page/map/map.component.ts:195]: Do we need `=== true` here?

**Review by gino-m — CHANGES_REQUESTED** [review]

**Comment by amysorto** [comment]
> Hi @amysorto! It looks like the geometry annotations (geometry task responses, non-LOI markers and polygons on the map) don't show their corresponding number shown in the panel. Want to file a TODO and a bug to do in a follow-up?
> 
> See designs here:
> 
> image

Tracking in #1477

**Review by gino-m — APPROVED** [review]

**Pull request merged**